src/MIDebugEngine/Engine.Impl/DebuggedProcess.cs

```diff
@@ -516,20 +516,8 @@ public DebuggedProcess(bool bLaunched, LaunchOptions launchOptions, ISampleEngin
                     }
                 }
 
-                var arch = await MICommandFactory.GetTargetArchitecture();
-                if (arch == TargetArchitecture.Unknown)
-                {
-                    if (LaunchOptions.TargetArchitecture != TargetArchitecture.Unknown)
-                    {
-                        arch = LaunchOptions.TargetArchitecture;
-                    }
-                    else
-                    {
-                        WriteOutput(ResourceStrings.Warning_UsingDefaultArchitecture);
-                        arch = TargetArchitecture.X64;  // use as default
-                    }
-                }
-                SetTargetArch(arch);
+                // Must wait to get the target architecture until the first break event.
+				EnqueueInternalBreakAction(ObtainTargetArchitecture);
```

Comment:
It wouldn't matter for vscode, but I'd rather see if I can get the correct fix sooner rather than later.